File: solvers/python/hello_json.py
import sys
import json
from src.utils import meaning_of_life

def solve(problem_id=None):
    problem_id = problem_id or (len(sys.argv) >= 2 and sys.argv[1])
    if not problem_id:
        print('Usage: python solver.py PROBLEM_ID')
        exit(1)

    with open(f'problems/{sys.argv[1]}.json', 'r') as file:
        problem = json.loads(file.read())

    print("Hello json!")
    print(json.dumps(problem, indent=2))

if __name__ == '__main__':
    solve()

File: solvers/cpp/src/hello_json.cpp
#include "shared/libs/json.hpp"

#include <fstream>
#include <iostream>

using json = nlohmann::json;

int main(int argc, char *argv[]) {
  if (argc < 2) {
    std::cout << "Usage: ./solver PROBLEM_ID" << std::endl;
    return 1;
  }

  std::string problem = std::string(argv[1]) + ".json";
  std::cout << "Problem: " << problem << std::endl;

  // read a JSON file
  std::ifstream i("problems/" + problem);
  json j;
  i >> j;

  // write prettified JSON to another file
  std::cout << std::setw(4) << j << std::endl;

  std::cout << "Value of `id` is: " << j["id"] << std::endl;

  return 0;
}
